chain-core/src/state/account/op/data/deposit.rs

```diff
@@ -0,0 +1,86 @@
+use crate::state::account::address::StakedStateAddress;
+use crate::state::account::op::data::attribute::StakedStateOpAttributes;
+use crate::tx::data::input::TxoPointer;
+use crate::tx::TransactionId;
+use parity_scale_codec::{Decode, Encode, Error, Input, Output};
+#[cfg(not(feature = "mesalock_sgx"))]
+use serde::{Deserialize, Serialize};
+#[cfg(not(feature = "mesalock_sgx"))]
+use std::fmt;
+
+/// Each input is 34 bytes
+///
+/// Assuming maximum inputs allowed are 64,
+/// So, maximum deposit transaction size (34 * 64) + 21 (address) + 1 (attributes) = 2198 bytes
+const MAX_DEPOSIT_TX_SIZE: usize = 2200; // 2200 bytes
+
+/// takes UTXOs inputs, deposits them in the specified StakedState's bonded amount - fee
+/// (updates StakedState's bonded + nonce)
+#[derive(Debug, PartialEq, Eq, Clone)]
+#[cfg_attr(not(feature = "mesalock_sgx"), derive(Serialize, Deserialize))]
+pub struct DepositBondTx {
+    pub inputs: Vec<TxoPointer>,
+    pub to_staked_account: StakedStateAddress,
+    pub attributes: StakedStateOpAttributes,
+}
+
+impl Decode for DepositBondTx {
+    fn decode<I: Input>(input: &mut I) -> Result<Self, Error> {
+        let size = input
+            .remaining_len()?
+            .ok_or_else(|| "Unable to calculate size of input")?;
+
+        if size > MAX_DEPOSIT_TX_SIZE {
+            return Err("Input too large".into());
+        }
+
+        let inputs = <Vec<TxoPointer>>::decode(input)?;
+        let to_staked_account = StakedStateAddress::decode(input)?;
+        let attributes = StakedStateOpAttributes::decode(input)?;
+
+        Ok(DepositBondTx {
+            inputs,
+            to_staked_account,
+            attributes,
+        })
+    }
+}
+
+impl Encode for DepositBondTx {
+    fn encode_to<EncOut: Output>(&self, dest: &mut EncOut) {
+        dest.push(&self.inputs);
```

Comment:
Do the transaction types need to support the future extensions?

Reply:
There are 2 ways:

1. If the extension is more or less the same behaviour, they already have that on their components -- e.g. if there's a new way to lock transaction output, it can be a variant in `ExtendedAddr` + a corresponding witness type.
(could be even to e.g. support a different signature scheme)

2. If the extension is a different behaviour, it'll be a new transaction type.

I can expand on that in a comment